doc/nrf/libraries/others/emds.rst

```diff
@@ -14,43 +14,62 @@ Implementation
 The :kconfig:option:`CONFIG_EMDS` Kconfig option enables the emergency data storage.
 
-The application must initialize the pre-allocated storage area by using the :c:func:`emds_init` function.
-The :kconfig:option:`CONFIG_EMDS_SECTOR_COUNT` option defines how many sectors should be used to store data.
-
-The allocated storage space must be larger than the combined data stored by the application.
+The EMDS is implemented to use the persistent memory in form of storage partitions, which can be either flash or RRAM.
+The static partition manager file :file:`pm.yml.emds` is required for defining the storage partitions.
+You must align the storage partitions to the write block size, which is 4 bytes for flash and 16 bytes for RRAM.
+The EMDS mandates at least two storage partitions to allow restoring of the last known copy of data.
+The application must initialize storage partitions by using the :c:func:`emds_init` function.
+
+The partition storage space for each partition must be larger than the combined data stored by the application.
+To find out how much storage space is needed, call the :c:func:`emds_store_size_get` function.
+Additionally, consider the size of the metadata, which is 32 bytes per snapshot.
 Allocating a larger storage area will demand more resources, but also increase the life expectancy of the persistent memory.
 The chosen size should reflect the amount of data stored, the available persistent memory and how the application calls the :c:func:`emds_store` function.
-In general, it should not be necessary to allocate a large storage area, since only a limited set of data should be stored to ensure swift completion of writing the data on shutdown.
 
 The memory location that is going to be stored must be added on initialization.
 All memory areas must be provided through entries containing an ID, data pointer, and data length, using the :c:func:`emds_entry_add` function and the :c:macro:`EMDS_STATIC_ENTRY_DEFINE` macro.
 Entries to be stored when the emergency data storage is triggered need their own unique IDs that are not changed after a reboot.
-
-When all entries are added, the :c:func:`emds_load` function restores the entries into the memory areas from the persistent memory.
-
-After restoring the previous data, the application must run the :c:func:`emds_prepare` function to prepare the storage area for receiving new entries.
-If the remaining empty storage area is smaller than the required data size, the storage area will be automatically erased to increase the available storage area.
+The entries are stored in the partition in form of snapshots.
+The snapshot is a packed copy of all entries that are registered with the EMDS library for storage in the data area of the partition.
+Each snapshot is described by the snapshot metadata that is stored in the metadata area of the partition.
+
+After initialization is completed by registering all EMDS entries, the application calls the :c:func:`emds_load` function.
```

Comment:
Consider splitting this paragraph to two or three shorter ones to improve readability.

Reply:
Okay.